ubuntu-checkpatch

```diff
@@ -362,8 +365,24 @@ class CheckCherrypickBackport(Check):
             commit = self.patch.backport
             self.result(PASS, "valid backport from commit: " + commit)
 
-        # TODO: Check provenance and upstream commit existence
-        return 0
+        if commit and self.patch.provenance in ("", "linux-next", "linux-stable"):
+            linux_base = "https://git.kernel.org/pub/scm/linux/kernel/git"
+            upstream = {
+                "": f"{linux_base}/torvalds/linux.git/commit/?id={commit}",
+                "linux-next": f"{linux_base}/stable/linux.git/commit/?id={commit}",
+                "linux-stable": f"{linux_base}/next/linux-next.git/commit/?id={commit}",
+            }[self.patch.provenance]
+            r = requests.get(upstream)
+            if r.status_code == 200:
+                self.result(PASS, "declared provenance contains referenced commit")
+            elif r.status_code == 404:
+                self.result(FAIL, "declared provenance does not contain referenced commit")
+                rc = 1
+            else:
+                self.result(WARN, f"unable to check upstream for commit: {upstream} returned HTTP_STATUS={r.status_code}")
```

Comment:
Warnings are supposed to be non-fatal so this should be FAIL.

Maybe we should introduce an offline mode that skips all checks that require an internet connection?

Reply:
Marked as FAIL.

Offline mode would be good. It might make sense to have a SKIP state too then. I'll work up another PR for that.